diff -uNr syncal-0.8.7/syncal.c syncal-0.8.7-newpisock/syncal.c
--- syncal-0.8.7/syncal.c	Thu Jan 10 18:17:15 2002
+++ syncal-0.8.7-newpisock/syncal.c	Wed Jul  3 17:56:17 2002
@@ -414,13 +414,13 @@
 		fflush( stdout);
 	}
 
-	if (!(pilot_sd = pi_socket(PI_AF_SLP, PI_SOCK_STREAM, PI_PF_PADP))){
+	if (!(pilot_sd = pi_socket(PI_AF_PILOT, PI_SOCK_STREAM, PI_PF_DLP))){
 		if ( !quiet )
 			fprintf(stderr, "pi_socket failed\n");
 		exit(1);
 	}
     
-	addr.pi_family = PI_AF_SLP;
+	addr.pi_family = PI_AF_PILOT;
 	strcpy(addr.pi_device, pilot_device);
 
 	if ( pi_bind(pilot_sd, (struct sockaddr*)&addr, sizeof(addr)) <0 ) {
diff -uNr syncal-0.8.7/syncal.h syncal-0.8.7-newpisock/syncal.h
--- syncal-0.8.7/syncal.h	Thu Jan 10 18:17:15 2002
+++ syncal-0.8.7-newpisock/syncal.h	Wed Jul  3 17:56:38 2002
@@ -8,7 +8,7 @@
  * See the file COPYING for details.
  */
 
-
+#include <errno.h>
 #include "pi-source.h"
 #include "pi-socket.h"
 #include "pi-todo.h"
